crates/core/src/db/datastore/locking_tx_datastore/datastore.rs

```diff
@@ -282,6 +282,10 @@ impl TxDatastore for Locking {
         tx.table_id_from_name(table_name, self.database_address)
     }
 
+    fn table_name_from_id_tx<'a>(&'a self, tx: &'a Self::Tx, table_id: TableId) -> Result<Option<Cow<'a, str>>> {
+        Ok(tx.table_exists(&table_id).map(Cow::Borrowed))
```

Comment:
Not sure if this is okay - the mut version does a db scan but this seems simpler but maybe it can be out of sync with the system table? idk, but this is fine for now.